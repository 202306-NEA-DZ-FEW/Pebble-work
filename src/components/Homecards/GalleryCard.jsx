import React, { useState, useEffect } from "react";
import styles from "@/styles/Homepage.module.css";

const GalleryCard = () => {
    const [isGallery, setIsGallery] = useState(false);
    const images = [
        "IMG1.png",
        "IMG2.png",
        "IMG3.png",
        "IMG4.png",
        "IMG5.png",
        "IMG6.png",
        "IMG7.png",
        "IMG8.png",
        "IMG9.png",
        "IMG10.png",
        "IMG11.png",
        "IMG12.png",
        "IMG13.png",
        "IMG14.png",
        "IMG15.png",
        "IMG16.png",
        "IMG17.png",
        "IMG18.png",
    ];
    useEffect(() => {
        const handleScroll = () => {
            // Get the current scroll position
            const scrollPosition = window.scrollY;
            // Get the height of the window
            const windowHeight = window.innerHeight;
            // Get the gallery element by its ID
            const element = document.getElementById("gallery");

            // Check if the element exists
            if (element) {
                // Get the position of the element relative to the top of the document
                const elementPosition = element.offsetTop;

                // Check if the scroll position is greater than half of the window height from the top of the element
                if (scrollPosition > elementPosition - windowHeight / 1.3) {
                    // Set the visibility of the featured card to true
                    setIsGallery(true);
                }
            }
        };

        // Add scroll event listener and call the handleScroll function
        window.addEventListener("scroll", handleScroll);

        // Clean up by removing the scroll event listener when the component is unmounted
        return () => {
            window.removeEventListener("scroll", handleScroll);
        };
    }, []);

    return (
        <div className='flex justify-center items-center shadow-md'>
            <div className='grid grid-cols-2 md:grid-cols-3 lg:grid-cols-6 gap-4'>
                {images.map((image, index) => (
                    <div
                        id='gallery'
                        key={index}
                        className={`bg-gray-100 xl:w-[190px] xl:h-[172px] lg:w-[156px] lg:h-[154px] m-2 sm:w-[138px] sm:h-[132px] w-[112px] h-[106px]  flex items-center justify-center ${
                            styles.gallery
                        } ${
                            isGallery ? styles.gallerySlide : styles.invisible
                        }`}
                        style={{
                            animationDelay: `${index * 0.1}s`,
                        }}
                    >
                        <img
                            style={{
                                objectPosition: "center",
                                objectFit: "cover",
                                display: "block",
                                width: "100%",
                                aspectRatio: "1/1",
                            }}
                            loading='lazy'
                            src={`/images/${image}`}
                            alt={`Image ${index + 1}`}
                        />
                    </div>
                ))}
            </div>
        </div>
    );
};

export default GalleryCard;
