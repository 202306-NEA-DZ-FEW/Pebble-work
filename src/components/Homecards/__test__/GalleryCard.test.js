import { render } from "@testing-library/react";
import React from "react";

import GalleryCard from "@/components/Homecards/GalleryCard";
import { useTranslation } from "next-i18next";

jest.mock("next-i18next", () => ({
    useTranslation: () => ({
        t: (str) => str,
        i18n: {
            language: "en",
        },
    }),
}));

describe("GalleryCard", () => {
    it("should match snapshot", () => {
        const { asFragment } = render(<GalleryCard />);
        expect(asFragment()).toMatchSnapshot();
    });
});
