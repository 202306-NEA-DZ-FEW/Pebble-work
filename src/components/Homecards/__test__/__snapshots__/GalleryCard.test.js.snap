// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GalleryCard should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex justify-center items-center p-4"
  >
    <div
      class="grid grid-cols-2 md:grid-cols-3 lg:grid-cols-6 gap-4"
    >
      <div
        class="bg-gray-100 xl:w-[190px] xl:h-[172px] lg:w-[156px] lg:h-[154px] m-2 sm:w-[138px] sm:h-[132px] w-[112px] h-[106px]  flex items-center justify-center gallery invisible"
        id="gallery"
        style="animation-delay: 0s;"
      >
        <img
          alt="Image 1"
          loading="lazy"
          src="/images/en/IMG1.png"
          style="object-position: center; object-fit: cover; display: block; width: 100%;"
        />
      </div>
      <div
        class="bg-gray-100 xl:w-[190px] xl:h-[172px] lg:w-[156px] lg:h-[154px] m-2 sm:w-[138px] sm:h-[132px] w-[112px] h-[106px]  flex items-center justify-center gallery invisible"
        id="gallery"
        style="animation-delay: 0.1s;"
      >
        <img
          alt="Image 2"
          loading="lazy"
          src="/images/en/IMG2.png"
          style="object-position: center; object-fit: cover; display: block; width: 100%;"
        />
      </div>
      <div
        class="bg-gray-100 xl:w-[190px] xl:h-[172px] lg:w-[156px] lg:h-[154px] m-2 sm:w-[138px] sm:h-[132px] w-[112px] h-[106px]  flex items-center justify-center gallery invisible"
        id="gallery"
        style="animation-delay: 0.2s;"
      >
        <img
          alt="Image 3"
          loading="lazy"
          src="/images/en/IMG3.png"
          style="object-position: center; object-fit: cover; display: block; width: 100%;"
        />
      </div>
      <div
        class="bg-gray-100 xl:w-[190px] xl:h-[172px] lg:w-[156px] lg:h-[154px] m-2 sm:w-[138px] sm:h-[132px] w-[112px] h-[106px]  flex items-center justify-center gallery invisible"
        id="gallery"
        style="animation-delay: 0.30000000000000004s;"
      >
        <img
          alt="Image 4"
          loading="lazy"
          src="/images/en/IMG4.png"
          style="object-position: center; object-fit: cover; display: block; width: 100%;"
        />
      </div>
      <div
        class="bg-gray-100 xl:w-[190px] xl:h-[172px] lg:w-[156px] lg:h-[154px] m-2 sm:w-[138px] sm:h-[132px] w-[112px] h-[106px]  flex items-center justify-center gallery invisible"
        id="gallery"
        style="animation-delay: 0.4s;"
      >
        <img
          alt="Image 5"
          loading="lazy"
          src="/images/en/IMG5.png"
          style="object-position: center; object-fit: cover; display: block; width: 100%;"
        />
      </div>
      <div
        class="bg-gray-100 xl:w-[190px] xl:h-[172px] lg:w-[156px] lg:h-[154px] m-2 sm:w-[138px] sm:h-[132px] w-[112px] h-[106px]  flex items-center justify-center gallery invisible"
        id="gallery"
        style="animation-delay: 0.5s;"
      >
        <img
          alt="Image 6"
          loading="lazy"
          src="/images/en/IMG6.png"
          style="object-position: center; object-fit: cover; display: block; width: 100%;"
        />
      </div>
      <div
        class="bg-gray-100 xl:w-[190px] xl:h-[172px] lg:w-[156px] lg:h-[154px] m-2 sm:w-[138px] sm:h-[132px] w-[112px] h-[106px]  flex items-center justify-center gallery invisible"
        id="gallery"
        style="animation-delay: 0.6000000000000001s;"
      >
        <img
          alt="Image 7"
          loading="lazy"
          src="/images/en/IMG7.png"
          style="object-position: center; object-fit: cover; display: block; width: 100%;"
        />
      </div>
      <div
        class="bg-gray-100 xl:w-[190px] xl:h-[172px] lg:w-[156px] lg:h-[154px] m-2 sm:w-[138px] sm:h-[132px] w-[112px] h-[106px]  flex items-center justify-center gallery invisible"
        id="gallery"
        style="animation-delay: 0.7000000000000001s;"
      >
        <img
          alt="Image 8"
          loading="lazy"
          src="/images/en/IMG8.png"
          style="object-position: center; object-fit: cover; display: block; width: 100%;"
        />
      </div>
      <div
        class="bg-gray-100 xl:w-[190px] xl:h-[172px] lg:w-[156px] lg:h-[154px] m-2 sm:w-[138px] sm:h-[132px] w-[112px] h-[106px]  flex items-center justify-center gallery invisible"
        id="gallery"
        style="animation-delay: 0.8s;"
      >
        <img
          alt="Image 9"
          loading="lazy"
          src="/images/en/IMG9.png"
          style="object-position: center; object-fit: cover; display: block; width: 100%;"
        />
      </div>
      <div
        class="bg-gray-100 xl:w-[190px] xl:h-[172px] lg:w-[156px] lg:h-[154px] m-2 sm:w-[138px] sm:h-[132px] w-[112px] h-[106px]  flex items-center justify-center gallery invisible"
        id="gallery"
        style="animation-delay: 0.9s;"
      >
        <img
          alt="Image 10"
          loading="lazy"
          src="/images/en/IMG10.png"
          style="object-position: center; object-fit: cover; display: block; width: 100%;"
        />
      </div>
      <div
        class="bg-gray-100 xl:w-[190px] xl:h-[172px] lg:w-[156px] lg:h-[154px] m-2 sm:w-[138px] sm:h-[132px] w-[112px] h-[106px]  flex items-center justify-center gallery invisible"
        id="gallery"
        style="animation-delay: 1s;"
      >
        <img
          alt="Image 11"
          loading="lazy"
          src="/images/en/IMG11.png"
          style="object-position: center; object-fit: cover; display: block; width: 100%;"
        />
      </div>
      <div
        class="bg-gray-100 xl:w-[190px] xl:h-[172px] lg:w-[156px] lg:h-[154px] m-2 sm:w-[138px] sm:h-[132px] w-[112px] h-[106px]  flex items-center justify-center gallery invisible"
        id="gallery"
        style="animation-delay: 1.1s;"
      >
        <img
          alt="Image 12"
          loading="lazy"
          src="/images/en/IMG12.png"
          style="object-position: center; object-fit: cover; display: block; width: 100%;"
        />
      </div>
      <div
        class="bg-gray-100 xl:w-[190px] xl:h-[172px] lg:w-[156px] lg:h-[154px] m-2 sm:w-[138px] sm:h-[132px] w-[112px] h-[106px]  flex items-center justify-center gallery invisible"
        id="gallery"
        style="animation-delay: 1.2000000000000002s;"
      >
        <img
          alt="Image 13"
          loading="lazy"
          src="/images/en/IMG13.png"
          style="object-position: center; object-fit: cover; display: block; width: 100%;"
        />
      </div>
      <div
        class="bg-gray-100 xl:w-[190px] xl:h-[172px] lg:w-[156px] lg:h-[154px] m-2 sm:w-[138px] sm:h-[132px] w-[112px] h-[106px]  flex items-center justify-center gallery invisible"
        id="gallery"
        style="animation-delay: 1.3s;"
      >
        <img
          alt="Image 14"
          loading="lazy"
          src="/images/en/IMG14.png"
          style="object-position: center; object-fit: cover; display: block; width: 100%;"
        />
      </div>
      <div
        class="bg-gray-100 xl:w-[190px] xl:h-[172px] lg:w-[156px] lg:h-[154px] m-2 sm:w-[138px] sm:h-[132px] w-[112px] h-[106px]  flex items-center justify-center gallery invisible"
        id="gallery"
        style="animation-delay: 1.4000000000000001s;"
      >
        <img
          alt="Image 15"
          loading="lazy"
          src="/images/en/IMG15.png"
          style="object-position: center; object-fit: cover; display: block; width: 100%;"
        />
      </div>
      <div
        class="bg-gray-100 xl:w-[190px] xl:h-[172px] lg:w-[156px] lg:h-[154px] m-2 sm:w-[138px] sm:h-[132px] w-[112px] h-[106px]  flex items-center justify-center gallery invisible"
        id="gallery"
        style="animation-delay: 1.5s;"
      >
        <img
          alt="Image 16"
          loading="lazy"
          src="/images/en/IMG16.png"
          style="object-position: center; object-fit: cover; display: block; width: 100%;"
        />
      </div>
      <div
        class="bg-gray-100 xl:w-[190px] xl:h-[172px] lg:w-[156px] lg:h-[154px] m-2 sm:w-[138px] sm:h-[132px] w-[112px] h-[106px]  flex items-center justify-center gallery invisible"
        id="gallery"
        style="animation-delay: 1.6s;"
      >
        <img
          alt="Image 17"
          loading="lazy"
          src="/images/en/IMG17.png"
          style="object-position: center; object-fit: cover; display: block; width: 100%;"
        />
      </div>
      <div
        class="bg-gray-100 xl:w-[190px] xl:h-[172px] lg:w-[156px] lg:h-[154px] m-2 sm:w-[138px] sm:h-[132px] w-[112px] h-[106px]  flex items-center justify-center gallery invisible"
        id="gallery"
        style="animation-delay: 1.7000000000000002s;"
      >
        <img
          alt="Image 18"
          loading="lazy"
          src="/images/en/IMG18.png"
          style="object-position: center; object-fit: cover; display: block; width: 100%;"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
